src/pluck.c

```diff
@@ -296,21 +298,29 @@ static int check_double_index_finiteness(double val, SEXP index, int i, bool str
 }
 
 static int check_double_index_length(double val, int n, int i, bool strict) {
-  if (val < 0) {
+  if (val == 0) {
+    if (strict) {
+      Rf_errorcall(R_NilValue, "Index %d is zero", i + 1);
```

Comment:
These error messages are missing full stops.

Reply:
It feels like we could do better but I'm not sure how. Maybe:

```
Error in `chuck()`:
! Problem with 3rd subscript.
Cause by error:
! Can't extract at location zero.
```

And

```
Error in `chuck()`:
! Problem with 3rd subscript.
Caused by error:
! Can't extract elements past the end.
i Location 10 doesn't exist, there are only 4 elements.

Error in `chuck()`:
! Problem with 3rd subscript.
Caused by error:
! Can't extract elements past the beginning.
i Negative location -10 doesn't exist, there are only 4 elements.
```

Reply:
See also #787